ext/rbs_extension/parser.c

```diff
@@ -963,25 +963,25 @@ static VALUE parse_simple(parserstate *state) {
     return type;
   }
   case kBOOL:
-    return rbs_base_type(RBS_Types_Bases_Bool, rbs_location_current_token(state));
```

Comment:
All 9 of these types used to have their own Ruby classes, but shared this same "constructor" (`rbs_base_type()`).

We will eventually want 9 different C structs to model these (to keep a simple 1:1 correspondence to the Ruby classes), with 9 different constructor functions. We did that here, since it simplifies automation.